src/compiler/utilitiesPublic.ts

```diff
@@ -1211,12 +1213,10 @@ function formatJSDocLink(link: JSDocLink | JSDocLinkCode | JSDocLinkPlain) {
  */
 export function getEffectiveTypeParameterDeclarations(node: DeclarationWithTypeParameters): readonly TypeParameterDeclaration[] {
     if (isJSDocSignature(node)) {
-        if (isJSDoc(node.parent)) {
-            const overloadTag = find(node.parent.tags, (tag) => {
-                return isJSDocOverloadTag(tag) && tag.typeExpression === node;
-            });
-            if (overloadTag) {
-                return flatMap(node.parent.tags, tag => isJSDocTemplateTag(tag) ? tag.typeParameters : undefined);
+        if (isJSDocOverloadTag(node.parent)) {

```

Comment:
Nice fix. Reviewing the old code, I thought JSDocSignature was a supertype of JSDocOverload. But it's actually a child property, so `node.parent` of a JSDocSignature isn't the containing JSDoc, but a JSDocOverloadTag.